lib/ball.js

```diff
@@ -23,50 +23,57 @@ let ball = class Ball {
     this.x += this.xSpeed;
     this.y += this.ySpeed;
     return this;
-  };
+  }
 
   wallCollision(canvas) {
-    //left or right
-    if(this.x + this.xSpeed > canvas.width - this.radius ||
-       this.x + this.xSpeed < 0 + this.radius) {
+    //left
```

Comment:
On this line, I didn't change how it was detected. I split them into separate conditions because their behavior is now slightly different. See the next line note for the real change.